tests/lib/cli-engine.js

```diff
@@ -2566,6 +2566,43 @@ describe("CLIEngine", () => {
                 });
             });
         });
+
+        describe("Patterns which match no file should throw errors.", () => {
+            beforeEach(() => {
+                engine = new CLIEngine({
+                    cwd: getFixturePath("cli-engine"),
+                    useEslintrc: false
+                });
+            });
+
+            it("one file", () => {
+                assert.throws(() => {
+                    engine.executeOnFiles(["non-exist.js"]);
+                }, "'non-exist.js' was not found.");
+            });
+
+            it("should not throw if the directory exists", () => {
+                engine.executeOnFiles(["empty"]);
```

Comment:
Could we `assert.doesNotThrow` just to be explicit about what we're asserting on? Thanks!

Reply:
I would, but I'll defer to consensus. I like `assert.doesNotThrow` for its explicitness if nothing else. But I realize I may be in the minority here.

Reply:
I suppose I'll add it now, and it can be removed later if we decide to do that.

Reply:
Actually, I just realized the question is moot because this should be throwing anyway. 😄